test/e2e_node/pod_conditions_test.go

```diff
@@ -137,20 +138,27 @@ func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, chec
 
 		ginkgo.By("update pod related volume resource to unblock sandbox creation")
 
-		e2epod.NewPodClient(f).Update(ctx, p.Name, func(pod *v1.Pod) {
-			pod.Spec.Volumes = []v1.Volume{
-				{
-					Name: "secret",
-					VolumeSource: v1.VolumeSource{
-						Secret: &v1.SecretVolumeSource{
-							SecretName: "secret",
+		secretName := "test-secret"
+		applyConfig := &corev1.PodApplyConfiguration{
+			Spec: &corev1.PodSpecApplyConfiguration{
+				Volumes: []corev1.VolumeApplyConfiguration{
+					{
+						Name: &secretName,
+						VolumeSourceApplyConfiguration: corev1.VolumeSourceApplyConfiguration{
+							Secret: &corev1.SecretVolumeSourceApplyConfiguration{
+								SecretName: &secretName,
+							},
 						},
 					},
 				},
-			}
+			},
+		}
+		e2epod.NewPodClient(f).Apply(ctx, applyConfig, metav1.ApplyOptions{})
+
```

Comment:
the original test was flaked due to I use update to create volume which is not allowed. using apply should be able to fix the flake

Reply:
Can you do this with `Kind` or a k8 cluster?  ie update a volume to an existing pod.  Usually pod updates are pretty locked down.